extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -492,7 +492,7 @@ export class VisibilityManagerForDoc extends VisibilityManager {
       return;
     }
     element.getLayoutBox = () => {
-      return this.viewport_.getRect();
+      return this.viewport_.getLayoutRect(element);
```

Comment:
So, I tried that. But I got confused with different aspects of offsetting: some worked in AMP->inabox, others in host->inabox. Generally, I changed the styling of the root element to make the size of root always the same as the size of the embed. So this should be, at least, possible...